daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala

```diff
@@ -221,75 +221,24 @@ private[lf] object Speedy {
       err: String
   ) extends RuntimeException(err, null, true, false)
 
-  @throws[SErrorDamlException]
   private[speedy] def buildDiscTable(
```

Comment:
As this function can now operate assuming that `Preprocessor#preprocessDisclosuredContracts` has successfully ran, we no longer need any validation logic within this method.